games/race/RaceEndGame.tsx

```diff
@@ -1,57 +1,98 @@
-import { Text, TouchableOpacity, View } from "react-native";
+import { Text, TouchableOpacity, FlatList, View } from "react-native";
 import mainStyles from "../../styles/MainStyles";
 import mainGamesStyles from "../../styles/games/MainGamesStyles";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import FrontLlamaCenter from "../../components/FrontLlamaCenter";
 import { NativeStackScreenProps } from "@react-navigation/native-stack";
-import Toast from "react-native-toast-message";
 import { NavigationProp, useNavigation } from "@react-navigation/native";
 import { MainStackParamList } from "../../App";
 import { RaceStackParamList } from "./RaceStack";
 import buttonGamesStyles from "../../styles/games/ButtonGamesStyles";
 import textGamesStyles from "../../styles/games/TextGamesStyles";
+import PlayerListItem, { PlaceItem } from "../../components/PlayerListItem";
 
 type Props = NativeStackScreenProps<RaceStackParamList, "EndGame">;
 type MainStack = NavigationProp<MainStackParamList, "Home">;
 
 function RaceEndGameScreen({ route }: Props) {
   const parentNavigation = useNavigation<MainStack>();
-  const { points, isWinner } = route.params;
+  const { scoreboard } = route.params;
+  const [final, setFinal] = useState<PlaceItem[]>();
 
-  async function exitGameHandler() {
-    try {
-      parentNavigation.navigate("Home");
-    } catch (error) {
-      console.error(error);
+  // const [yourPlace, setYourPlace] = useState<number>();
+  // fix when user will be added
+
+  function exitGameHandler() {
+    parentNavigation.navigate("Home");
+  }
+
+  function handleData(): PlaceItem[] {
+    const final = [];
+
+    let currentPlace = 0;
+    let currentPoints = 0;
+    for (const { username, points } of scoreboard) {
```

Comment:
you can use scoreboard.map() function here to return new array without creating new `final` array and pushing stuff there

Reply:
will be changed after adding user :D